File: development/NET/samples/client/AEBrowse/VB/OutProc.vb
Namespace AEBrowse
	''' <summary>
	''' Summary description for OutProc.
	''' </summary>
	Public Class OutProc
#Region "Private Attributes"
		'-------------------------
		Private Shared m_opcClient As OpcClient = Nothing

		'--
#End Region

#Region "Public Properties"
		'------------------------

		Public ReadOnly Property OpcClient() As OpcClient
			Get
				Return m_opcClient
			End Get
		End Property
		'--
#End Region

#Region "Public Methods"
		'--------------------------
		Public Sub CreateOpcClient()
			If m_opcClient Is Nothing Then
				m_opcClient = New OpcClient
			End If
		End Sub
		'--
#End Region

	End Class
End Namespace

File: development/NET/test_apps/C#/COM TestServer/OutProc.cs
using System;
using System.Threading;
using System.IO;

namespace TestServer
{
	/// <summary>
	/// Summary description for OutProc.
	/// </summary>
	public class OutProc
	{
		#region Private Attributes
		//-------------------------
		private static OpcServer m_opcServer = null;

		//--
		#endregion

		#region Public Properties
		//------------------------

		public OpcServer OpcServer
		{
			get
			{
				return m_opcServer;
			}	//	end get
		}	//	end OpcServer
		//--
		#endregion

		#region Public Methods
		//--------------------------
		public void CreateOpcServer()
		{
			if (m_opcServer == null)
			{
				m_opcServer = new OpcServer();
			}	//	em=nd if
		}	//	end CreateOpcClient
		//--
		#endregion

	}	//	end class Outproc
}	//	end namespace
